Gallery-Server/src/main/java/edu/yu/cs/gallery/Utility.java

```diff
@@ -22,35 +23,77 @@
 public class Utility {
     protected Gallery gallery;
     protected Map<Long, URL> allServers;
+    protected Long leaderID;
     
     @Inject
     GalleryRepository gr;
 
-    public Response redirect (long id, @Context UriInfo uriInfo) throws URISyntaxException {
+    //Forwards the URL to the client with a 307
+    protected Response temporaryRedirect(long id, @Context UriInfo uriInfo) throws URISyntaxException {
         if (allServers.keySet().contains(id)) {
-            return Response.temporaryRedirect(new URI (allServers.get(id).toString() + uriInfo.getPath())).build();
+            return Response.temporaryRedirect(new URI(allServers.get(id).toString() + uriInfo.getPath())).build();
         }
         return Response.status(NOT_FOUND).build();
     }
     
-    public Response redirect (long[] galleries, @Context UriInfo uriInfo) throws URISyntaxException {
+    //URI must contain the string "batch" in place of the galleryId 
+    @SuppressWarnings("all")
```

Comment:
this is usually not a great sign... 
what warnings did you have to supress?

Reply:
![image](https://user-images.githubusercontent.com/98489038/180456838-e73ce1d6-a697-4589-9e3b-a1e6ca74e962.png)

Just these errors from the cast. Thought it was a bit annoying to see so suppressed it. do you have an issue with this use case?

Reply:
In a real world scenario I'd probably insist on getting rid of the warnings, but its ok for now.

Reply:
How would you get rid of those warnings here?

Reply:
I'd have to check the exact nature of the warning, but you would need to do some casting, and adding type parameters looks like.